titus-ext/jooq/src/main/java/com/netflix/titus/ext/jooq/relocation/JooqTaskRelocationResultStore.java

```diff
@@ -146,6 +148,24 @@ private void createSchemaIfNotExist() {
         });
     }
 
+    /**
+     * Removes all entries older than the given time threshold.
+     */
+    int removeExpiredData(long timeThreshold) {
```

Comment:
I think this method should first do a count based on an index to see how many items there are matching the criteria to be deleted and then delete a configured amount of them. That way you can set a metric that shows how many items are eligible for deletion versus are being deleted in the case of large spikes. Also, you can write a better log debug statement saying which items are explicitly being deleted in each batch so it is easier to troubleshoot.